sycl/doc/design/GlobalObjectsInRuntime.md

```diff
@@ -22,7 +22,7 @@ like in example below:
 sycl::queue Queue;
 
 int main() {
-  Queue = sycl::queue{sycl::default_selector{}.select_device()};
+  Queue = sycl::queue{sycl::default_selector_v};
```

Comment:
This example might not be best suited here anymore. `default_selector_v` is a function in our implementation so there is no "race" between it and `Queue` object created at line 22.

Line 25 is also useless now as the `Queue` is already initialized to the default device.

Reply:
Agree about Line 25, but I think we should leave it. The whole text is about how you get undefined behavior when using global objects because the C++ standard does not specify creation/destruction order, and how the example _seems_ syntactically correct, yet is still undefined behavior.  If we remove 25 then the surrounding text isn't really making a point.

Reply:
Where is UB here now?

Reply:
the queue is created outside main. The "seems syntactically correct" part is declaring the queue outside main but initializing it inside main.  If we remove line 25 there is no "seems syntactically correct" part.  Read the entire section and you'll see what I mean.

Reply:
probably the whole thing should be re-written, but that's outside the scope of this PR. I'm just trying to bring the example up to date, not address its overall point.

Reply:
> part is declaring the queue outside main but initializing it inside main

I don't think it's true. We do create and define a queue outside main. Then we override the value inside the variable with another one, while the original is being destroyed.

Reply:
correct. but to make that point, we need to keep line 25.

Reply:
Still, where is UB?

Reply:
https://github.com/intel/llvm/blob/9f5a22372161a18d0adf05dabfa22ff49f8d083f/sycl/doc/design/GlobalObjectsInRuntime.md